### DISCUSSION TITLE
Energy Dashboard colors for individual device

### DISCUSSION THREAD

**Discussion by mspcoding** [home-assistant/frontend#22418] [Dashboard]
Why does the colors of all individual devices in the energy dashboard changes after each HA update?
Please add a user color selection for each individual device to set the desired color for a specific device. Currently it's just random and looks terrible.

**Comment by karwosts** [comment]
It does not change "every update". It has changed once for 2024.10 for https://github.com/home-assistant/frontend/pull/21839

If you don't like the colors, you are free to define your own colors in a theme with --graph-color-1, --graph-color-2, etc...

**Reply by lucascosti** [reply]
@karwosts IIUC, using `--graph-color-1` etc changes general graph colors in HA outside of the Energy dashboard too though (which isn't really desirable).

Having energy-specific graph colors for individual devices on the energy tab would be really helpful 🙇‍♂️

**Comment by mspcoding** [comment]
<img width="1163" alt="Bildschirmfoto 2024-10-18 um 20 41 50" src="https://github.com/user-attachments/assets/2f19ecf5-5936-42c6-9f86-ea423232fdba">

But how can I assign an own defined color to a specific individual device in the energy dashboard?
Similar devices, like washer and dryer should be in the same color range, so I want to assign a specific color to an individual device.

**Comment by HM404** [comment]
Hi,
I have exactly the same intent. I found out, that the colors respond with the order in which the devices are configured in the list, but it would be much easier to choose or define a specific color for each device individually. 
I found the list of first 10 colors in #21839
